tvm_utility/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(tvm_utility)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+find_package(autoware_build_flags REQUIRED)
+
+# set dependencies for the tests
+set(CATKIN_TEST_DEPENDS pcl_ros autoware_msgs)
+
+find_package(
+  catkin REQUIRED
+  COMPONENTS ${CATKIN_TEST_DEPENDS}
+)
+
+# declare dependency packages to catkin
+catkin_package(
+  INCLUDE_DIRS include
+  CATKIN_DEPENDS ${CATKIN_TEST_DEPENDS}
+)
+
+include_directories(
+  include
+  ${catkin_INCLUDE_DIRS}
+)
+
+if(CATKIN_ENABLE_TESTING)
+  find_package(rostest REQUIRED)
+
+  # compile each folder inside test/ as a test case
+  file(GLOB test_cases test/*)
+  foreach(test_folder ${test_cases})
+    if(EXISTS "${test_folder}/inference_engine_tvm_config.hpp"
+       AND EXISTS "${test_folder}/deploy_lib.so"
+       AND EXISTS "${test_folder}/deploy_graph.json"
+       AND EXISTS "${test_folder}/deploy_param.params"
+    )
+      # the folder name becomes the test case name
+      file(RELATIVE_PATH test_case_name ${CMAKE_CURRENT_LIST_DIR}/test ${test_folder})
+
+      # add all cpp files in the folder to the target
+      file(GLOB test_case_sources ${test_folder}/*.cpp)
+      catkin_add_gtest(${test_case_name} ${test_case_sources})
+
+      # set extra properties of the target
+      set_property(TARGET ${test_case_name} PROPERTY CXX_STANDARD 17)
+      target_link_libraries(${test_case_name}
+        ${catkin_LIBRARIES}
+        boost_system
+        dl
+        pthread
+        tvm_runtime
+        opencv_core
+        opencv_highgui
+        opencv_imgproc
+        opencv_imgcodecs
```

Comment:
These must all be `find_package()`'d.

Reply:
Done.

Reply:
When I said "all", I meant all packages in this list, not just the OpenCV ones. Every dependency you use in a ROS package must be available as a `rosdep` key (see https://github.com/ros/rosdistro/blob/master/rosdep/base.yaml for the list) which can be added to `package.xml` and must also be `find_package()`-able by CMake. The Boost::System and OpenCV dependencies can be added to `package.xml` with:

```
  <test_depend>libopencv-dev</test_depend>
  <test_depend>libboost-system-dev</test_depend>
```

But I'm not sure how you intend to get `tvm_runtime` or `dl` (what is `dl`?).

Reply:
dl is a standard machine learning buffer format that tvm uses.

tvm_runtime or dl does not have pre-compiled packages for any distributions. Hence this does not seem to fit into the rosdep paradigm.

@JWhitleyWork what do you suggest is the best approach in this instance?

Reply:
@LiyouZhou The typical way to approach this in the ROS ecosystem is by creating a ROS wrapper package. An example of one we've done before is [qpoases_vendor](https://github.com/Autoware-AI/qpoases_vendor). I'll start working on this for ROS1 Melodic and ROS2 Dashing. Unfortunately, the release process takes a little while (2-4 weeks) and we can't accept this MR until the package is available.

Reply:
This is being worked on below. Resolving conversation.